cmd/documentation/documentation.go

```diff
@@ -0,0 +1,435 @@
+		dirList,
+		tmpl,
+		contributors,
+		*verbose,
+		&config})
+	if err != nil {
+		log.Fatalf("Documentation Generation Tool - UpdateDocumentation error %s",
+			err)
+	}
+
+	fmt.Println("\nDocumentation Generation Tool - Finished")
+}
+
+// GetConfiguration retrieves the documentation configuration
+func GetConfiguration() (Config, error) {
+	var c Config
+	file, err := os.OpenFile("config.json", os.O_RDWR, os.ModePerm)
+	if err != nil {
+		fmt.Println("Creating configuration file, please add github repository path and preferences")
+
+		file, err = os.Create("config.json")
+		if err != nil {
+			return c, err
+		}
+
+		// Set default params for configuration
+		c.GithubRepo = DefaultRepo
+		c.ContributorFile = true
+		c.LicenseFile = true
+		c.RootReadme = true
+		c.ReferencePathToRepo = "../../"
+		c.Exclusions.Directories = []string{".github"}
+
+		data, mErr := json.MarshalIndent(c, "", " ")
+		if mErr != nil {
+			return c, mErr
+		}
+
+		_, err = file.WriteAt(data, 0)
+		if err != nil {
+			return c, err
+		}
+	}
+
+	defer file.Close()
+
+	config, err := ioutil.ReadAll(file)
+	if err != nil {
+		return c, err
+	}
+
+	err = json.Unmarshal(config, &c)
+	if err != nil {
+		return c, err
+	}
+
+	if c.GithubRepo == "" || c.GithubRepo == DefaultRepo {
+		return c, errors.New("repository not set in config.json file, please change")
+	}
+
+	if c.ReferencePathToRepo == "" {
+		return c, errors.New("reference path not set in the config.json file, please set")
+	}
+
+	return c, nil
+}
+
+// IsExcluded returns if the file path is included in the exclusion list
+func IsExcluded(path string, exclusion []string) bool {
+	for _, data := range exclusion {
+		if strings.Contains(path, data) {
+			return true
+		}
+	}
+	return false
+}
+
+// GetProjectDirectoryTree uses filepath walk functions to get each individual
+// directory name and path to match templates with
+func GetProjectDirectoryTree(c *Config, verbose bool) ([]string, error) {
+	var directoryData []string
+	if c.RootReadme { // Projects root README.md
+		directoryData = append(directoryData, c.ReferencePathToRepo)
+	}
+
+	if c.LicenseFile { // Standard license file
+		directoryData = append(directoryData, c.ReferencePathToRepo+LicenseFile)
+	}
+
+	if c.ContributorFile { // Standard contributor file
+		directoryData = append(directoryData, c.ReferencePathToRepo+ContributorFile)
+	}
+
+	walkfn := func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			// Bypass what is contained in config.json directory exclusion
+			if IsExcluded(info.Name(), c.Exclusions.Directories) {
+				if verbose {
+					fmt.Println("Excluding Directory:", info.Name())
+				}
+				return filepath.SkipDir
+			}
+			// Don't append parent directory
+			if strings.EqualFold(info.Name(), "..") {
+				return nil
+			}
+			directoryData = append(directoryData, path)
+		}
+		return nil
+	}
+
+	return directoryData, filepath.Walk(c.ReferencePathToRepo, walkfn)
```

Comment:
Could you please try running `documentation --v`. I had the same issue but I've been able to exclude it in my config but I'm going to do a PR which adds some more to the default list and uses compare instead of contains to be more specific since "web" would exclude "websocket"

```
Excluding Directory: .git
Excluding Directory: .github
Excluding Directory: .vscode
Excluding Directory: node_modules
```

Reply:
Funnily enough, on the failure without any exclusion I got this crazy path `Documentation Generation Tool - GetProjectDirectoryTree error CreateFile ..\..\web\node_modules\webdriver-js-extender\built\built\built\built\built\built\built\built\built\built\built\built\built\built\built\built\spec\command_tests\totally_real_apk.apk: The system cannot find the path specified.`

Reply:
I've set this `c.Exclusions.Directories = []string{".github", "node_modules"}` and it doesn't reach the part to be able to exclude the directories because I think its a fault in filepath.Walk + node_modules + windows, so it can't event reach the code which allows node_modules to be skipped properly 🤷‍♂ 
But if you got it working after an `npm install` then I'll test it after your commit 👍

Reply:
#387 Should fix this please review.